### PR TITLE
Cleanup imports in common.py, lstm.py, extract exceptions

### PR DESCRIPTION
**Pull request by kba**
* Move exceptions to its own module
* Make all imports explicit, so static code analysis tools like pyflakes can help
* remove redundant deprecated decorators
* Remove old load_components
* Remove old draw_pseg/draw_aligned

### CODE REVIEW

File: ocrolib/common.py [hunk 1]
```diff
@@ -11,97 +11,32 @@
 import warnings
 import inspect
 import glob
+import pickle
 import cPickle
+from ocrolib.exceptions import (BadClassLabel, BadInput, FileNotFound,
```
Comment (QuLogic) [inline ocrolib/common.py]: Should move down to third block (first-party imports).

File: ocrolib/common.py [hunk 2]
```diff
@@ -11,97 +11,32 @@
...
+                                OcropusException)
 
 import numpy
-from numpy import *
+from numpy import (amax, amin, array, bitwise_and, clip, dtype, mean, minimum,
+                   nan, ndarray, sin, sqrt, zeros)
 import pylab
-from pylab import imshow
-from scipy.ndimage import morphology,measurements
+from pylab import (clf, cm, ginput, gray, imshow, ion, subplot, where)
```
Comment (QuLogic) [inline ocrolib/common.py]: I've got a branch that removes `pylab` imports; should I open a PR for that?

File: ocrolib/common.py [hunk 3]
```diff
@@ -11,97 +11,32 @@
...
 import glob
+import pickle
 import cPickle
+from ocrolib.exceptions import (BadClassLabel, BadInput, FileNotFound,
+                                OcropusException)
 
 import numpy
-from numpy import *
+from numpy import (amax, amin, array, bitwise_and, clip, dtype, mean, minimum,
+                   nan, ndarray, sin, sqrt, zeros)
```
Comment (QuLogic) [inline ocrolib/common.py]: The branch also standardizes these imports too...

File: ocrolib/common.py [hunk 4]
```diff
@@ -11,97 +11,32 @@
...
 from default import getlocal
-from toplevel import *
+from toplevel import (checks, ABINARY2, AINT2, AINT3, BOOL, DARKSEG, GRAYSCALE,
+                      LIGHTSEG, LINESEG, PAGESEG)
 import chars
 import codecs
 import ligatures
 import lstm
 import morph
 import multiprocessing
```
Comment (QuLogic) [inline ocrolib/common.py]: Should move up to first block (standard library imports).

**Comment by zuphilip** [comment]
I see that the exceptions are deleted in `ocrolib/common.py` but I don't see another module with them. Is that (new) file missing here?

File: ocrolib/common.py (hunk 2 above)
Comment (kba) [inline ocrolib/common.py]: You mean https://github.com/QuLogic/ocropy/commits/standard-import? Sure, open a PR with the pertinent commits, looks really helpful, I'll happily merge/cherry-pick.

Comment (QuLogic) [inline ocrolib/common.py]: Yes, that's the one; though I just noticed that it's based on the Python 3 work, so it'll be a pain to cherry-pick. Should get that in ASAP and then the rest would be easier.

**Pull request merged**